backend/src/resolvers/user_mutations.js

```diff
@@ -0,0 +1,40 @@
+const hashing = require("../services/hashing.js");
+const jwt = require("../services/jwt.js");
+import { AuthenticationError, UserInputError } from "apollo-server";
+
+function isPasswordStrong(password) {
+  return password.length >= 8;
+}
+
+export const usersMutationResolver = {
+  Mutation: {
+    login: async (parent, args, context) => {
+      let user = await context.dataSources.db.getUserByEmail(args.email);
+      if (!user)
+        throw new UserInputError(
+          "There is no user registered with this Email!"
+        );
+      let isPasswordCorrect = await hashing.compare(
+        args.password,
+        user.password
+      );
+      if (!isPasswordCorrect)
+        throw new AuthenticationError("Password did not match!");
+      return jwt.issueToken(user.id);
+    },
+
+    signup: async (parent, args, context) => {
+      let emailExists = await context.dataSources.db.emailExists(args.email);
+      if (emailExists) throw new UserInputError("Email already exists!");
+      if (!isPasswordStrong(args.password))
+        throw new UserInputError("Password must be atleast 8 characters!");
+      let encryptedPassword = await hashing.hash(args.password);
+      let user = await context.dataSources.db.addNewUser({
+        name: args.name,
+        email: args.email,
+        password: encryptedPassword,
+      });
+      return jwt.issueToken(user.id);
+    },
+  },
```

Comment:
Meiner Einschätzung nach habt ihr alle Kriterien für das Login und den SignUp erfüllt. Ihr prüft, dass das Passwort mindestenz 8 Zeichen lang ist, dass Emails nicht doppelt verwendet werden, ihr verwendet Hashing (backend/src/services/hashing.js) und es sind keine unnötigen Informationen in den zurückgegebenen Tokens ( backend/src/services/jwt.js). 
An sich gefällt mit auch, dass ihr viele kleinere Datein verwendet, aber ich bin mir nicht sicher, ob das Auslagern des Hashings nicht schon zu viel des guten ist.